bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/handler/MeaterBridgeHandler.java

```diff
@@ -0,0 +1,166 @@
+ */
+package org.openhab.binding.meater.internal.handler;
+
+import static org.openhab.binding.meater.internal.MeaterBindingConstants.*;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.meater.internal.MeaterBridgeConfiguration;
+import org.openhab.binding.meater.internal.api.MeaterRestAPI;
+import org.openhab.binding.meater.internal.discovery.MeaterDiscoveryService;
+import org.openhab.binding.meater.internal.dto.MeaterProbeDTO;
+import org.openhab.core.i18n.LocaleProvider;
+import org.openhab.core.i18n.TranslationProvider;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link MeaterBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jan Gustafsson - Initial contribution
+ */
+@NonNullByDefault
+public class MeaterBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(MeaterBridgeHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_BRIDGE);
+
+    private final Gson gson;
+    private final HttpClient httpClient;
+    private final TranslationProvider i18nProvider;
+    private final LocaleProvider localeProvider;
+    private final Map<String, MeaterProbeDTO.Device> meaterProbeThings = new ConcurrentHashMap<>();
+
+    private int refreshTimeInSeconds = 300;
+    private @Nullable MeaterRestAPI api;
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    public MeaterBridgeHandler(Bridge bridge, HttpClient httpClient, Gson gson, TranslationProvider i18nProvider,
+            LocaleProvider localeProvider) {
+        super(bridge);
+        this.httpClient = httpClient;
+        this.gson = gson;
+        this.i18nProvider = i18nProvider;
+        this.localeProvider = localeProvider;
+    }
+
+    @Override
+    public void initialize() {
+        MeaterBridgeConfiguration config = getConfigAs(MeaterBridgeConfiguration.class);
+
+        api = new MeaterRestAPI(config, gson, httpClient, localeProvider);
+        refreshTimeInSeconds = config.refresh;
+
+        if (config.email.isEmpty() || config.password.isEmpty()) {
```

Comment:
It would be even better if you use `isBlank` instead of `isEmpty`

Reply:
Fixed!